### PR TITLE
Refactor eventbus and distsql handler

### PR DESCRIPTION
**Pull request by juaby**
Fixes #ISSUSE_ID.

Changes proposed in this pull request:
- Added event bus CompletableEventService and CompletableEvent proxy layer to intercept consumption events uniformly
- Added event AbstractBackendHandler, unified exception handling
- Fixed registry center adding data sources using unsupported Map data structures

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/eventbus/CompletableEventService.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+     */
+    @Subscribe
+    public void handle(final CompletableEvent completableEvent) {
+        try {
+            Method handler = targetMethods.get(completableEvent.getTarget().getClass());
+            if (null != handler) {
+                handler.invoke(target, completableEvent.getTarget());
+            }
+            completableEvent.getCompletableFuture().complete(true);
+        } catch (IllegalAccessException | InvocationTargetException e) {
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/eventbus/CompletableEventService.java:61]: final one?

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/eventbus/ShardingSphereEventBus.java [hunk 2]
```diff
@@ -20,23 +20,75 @@
...
         return ShardingSphereEventBusHolder.INSTANCE;
     }
     
+    /**
+     * Registers all subscriber methods on {@code object} to receive events.
+     *
+     * @param target whose subscriber methods should be registered.
+     * @param <T> subscriber type.
+     */
+    public <T> void register(final T target) {
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/eventbus/ShardingSphereEventBus.java:57]: Where is this function called?

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/eventbus/CompletableEventService.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    private final Map<Class<?>, Method> targetMethods;
+
+    public CompletableEventService(final T target) {
+        this.target = target;
+        this.targetMethods = Arrays.stream(target.getClass().getDeclaredMethods()).filter(method -> {
+            Class<?>[] parameterTypes = method.getParameterTypes();
+            return parameterTypes.length == 1 && null != method.getDeclaredAnnotation(Subscribe.class);
+        }).collect(Collectors.toMap(method -> {
+            Class<?>[] parameterTypes = method.getParameterTypes();
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/eventbus/CompletableEventService.java:43]: Is it supposed to remove all `@Subscribe` annotation from `renew()` of target classes?

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/eventbus/CompletableEventService.java (hunk 1 above)
Comment (juaby) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/eventbus/CompletableEventService.java]: Fixed

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/eventbus/CompletableEvent.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import lombok.RequiredArgsConstructor;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Completable event.
+ */
+@Getter
+@RequiredArgsConstructor
+public class CompletableEvent<E> {
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/eventbus/CompletableEvent.java:30]: final one?

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/eventbus/CompletableEvent.java (hunk 4 above)
Comment (juaby) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/eventbus/CompletableEvent.java]: Fixed

**Pull request closed without merging**